src/lib/components/forms/label-input.svelte

```diff
@@ -1,31 +1,54 @@
 <script>
+  export let value = '';
   export let label = '';
   export let id = '';
   export let type = 'text';
   export let placeholder = '';
   export let isTextArea = false;
+  export let required = false;
   export let horizontal = false;
+  export let textGap;
+  export let name = '';
 </script>
 
-<div class="flex flex-{horizontal ? 'row' : 'col'}">
+<div
+  class="flex flex-{horizontal ? 'row justify-between' : 'col'} {$$props.className}"
+  class:align-middle={horizontal && !isTextArea}
+>
   {#if label}
-    <label class="m-1 font-source_code font-bold text-white" for={id}>{label}</label>
+    <div style="width: {textGap}vw;">
+      <label
+        class="m-1 flex flex-row font-source_code font-bold text-white"
+        class:flex-row={required}
+        for={id}
+      >
+        {label}
+        {#if required}
+          <span class="text-sm">*</span>
+        {/if}
+      </label>
+    </div>
   {/if}
   {#if isTextArea}
     <textarea
       aria-label="textarea-input"
-      class="mb-2 min-h-[100px] w-full rounded-lg bg-white p-2 font-source_code text-primary placeholder-primary"
+      class="min-h-[100px] w-full rounded-lg bg-white p-2 font-source_code text-primary placeholder-primary"
```

Comment:
```suggestion
      class="w-full rounded-lg bg-white p-2 font-source_code text-primary placeholder-primary"
```
I don't think this is needed